src/python_routing_v02/compute_nhd_routing_SingleSeg_v02.py

```diff
@@ -1286,7 +1286,37 @@ def main():
                 ],
                 copy=False,
             )
-
+            
+        if output_parameters.get("write_wrf_hydro_channel_restart", None):
+            # directory containing WRF Hydro restart files
+            restart_dir = output_parameters.get("wrf_hydro_channel_restart_directory",None)
+ 
+            # list of WRF Hydro restart files
+            restart_files = glob.glob(
+                os.path.join(
+                    restart_dir, 
+                    output_parameters["wrf_hydro_channel_restart_pattern_filter"] + "[!.TROUTE]"
+                )
+            )  
+            
+            try:
+                if not pathlib.Path(restart_dir).exists() or len(restart_files) == 0:
+                    raise AssertionError
+            except AssertionError:
+                print('WRF Hydro restart files not found - Aborting restart write sequence')
+            else:
```

Comment:
This is a lame attempt at exception handling. I didn't want the whole workflow to fail if the restart files are not found. But maybe we want the whole process to fail?